### PR TITLE
NavBar component

### PR DESCRIPTION
**Pull request by strawberrie68**
NavBar renders the NavBar with icons and links

navbar pic

### CODE REVIEW

File: src/components/NavBar/NavBar.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+const NavBar = () => {
+  return (
+    <div className="h-full w-24 border bl-black-200 pt-10 icons-center flex flex-col">
+      <img src={logo} alt="logo" className="h-20 w-auto p-5" />
+
+      <div className="mt-16 p-2">
+        {navigationIcons.map((icon) => (
+          <NavLink key={icon.name} to={icon.href}>
+            {({isActive}) => (
+              <div className="flex flex-col items-center	mb-6">
```
Comment (laurawong) [inline src/components/NavBar/NavBar.jsx]:
```suggestion
              <div className="flex flex-col items-center mb-6">
```
Remove extra space

File: src/tests/NavBar.test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import "@testing-library/jest-dom";
+import {render, screen} from "@testing-library/react";
+import NavBar from "../components/NavBar/NavBar";
+import {BrowserRouter as Router} from "react-router-dom";
+
+describe("NavBar", () => {
+  test("renders NavBar text", () => {
+    render(
+      <Router>
+        <NavBar />
```
Comment (laurawong) [inline src/tests/NavBar.test.js:11]:
```suggestion
        <NavBar/>
```
Remove extra spaces in this PR please

File: src/tests/NavBar.test.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      </Router>
+    );
+    expect(screen.getByText("Home")).toBeInTheDocument();
```
Comment (laurawong) [inline src/tests/NavBar.test.js:14]: You probably should assert all the other navbar items.

File: src/components/NavBar/NavIconData.jsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  MagnifyingGlass,
+  Binoculars,
+  Book,
+  Archive,
+} from "@phosphor-icons/react";
+
+export const navigationIcons = [
+  {
+    name: "Home",
+    activeIcon: <House size={24} color="#000000" weight="fill" />,
```
Comment (laurawong) [inline src/components/NavBar/NavIconData.jsx]: Set 24 as a constant.

Comment (laurawong) [inline src/components/NavBar/NavIconData.jsx]: Set the colors #000000 and #9E9E9E as constants.

**Comment by strawberrie68** [comment]
navbar pic

File: src/components/NavBar/NavBar.jsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+const NavBar = () => {
+  return (
+    <div className=" icons-center bl-black-200 flex h-full w-20 flex-col border pt-10">
+      <img src={logo} alt="logo" className="h-20 w-auto p-5" />
+
+      <div className="mt-16 p-2">
+        {navigationIcons.map((icon) => (
+          <NavLink key={icon.name} to={icon.href}>
+            {({ isActive }) => (
```
Comment (laurawong) [inline src/components/NavBar/NavBar.jsx:13]: Where is `isActive` defined? Can you add a test for `isActive`?

Comment (strawberrie68) [reply src/components/NavBar/NavBar.jsx:13]: isActive is a built-in variable in NavLink!  Will do!

File: src/components/NavBar/NavBar.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+              <div
+                id={`${
+                  icon.name.split(" ").length > 1
+                    ? icon.name.replace(" ", "-")
+                    : icon.name
+                }-icon`}
```
Comment (laurawong) [inline src/components/NavBar/NavBar.jsx]: Was this added just for testing? Probably not worth adding. It looks confusing.

File: cypress/compoment/NavBar.cy.jsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
+import NavBar from "../../src/components/NavBar/NavBar";
+import { BrowserRouter as Router } from "react-router-dom";
+
+describe("Render <NavBar /> component", () => {
```
Comment (laurawong) [inline cypress/compoment/NavBar.cy.jsx]: I don't think this test suite is very helpful.

1. There is so much code duplication. Imagine that the color changes for some reason, you would have to rewrite the tests. Not maintainable.
2. The colors don't even correspond to the actual code. How am I supposed to know what `rgb(163, 163, 163)` is at first glance?
3. You added extra classes with confusing logic. It took me a while to understand what you were doing.

Something like this instead...
```
  it("icon is inactive by default", () => {
      cy.get('icon').should('have.class', 'icon-inactive');
  });

it("icon is active when clicked", () => {
      cy.get('icon')
      .click()
      .should('have.class', 'icon-active');
  });
```

File: src/components/NavBar/NavBar.jsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+                className="mb-6 flex flex-col	items-center"
+              >
+                {isActive ? icon.activeIcon : icon.inactiveIcon}
+                <p
+                  id={`${
```
Comment (laurawong) [inline src/components/NavBar/NavBar.jsx]: not needed

File: src/components/NavBar/NavBar.jsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+        {navigationIcons.map((icon) => (
+          <NavLink key={icon.name} to={icon.href}>
+            {({ isActive }) => (
+              <div
+                id={`${
+                  icon.name.split(" ").length > 1
+                    ? icon.name.replace(" ", "-")
+                    : icon.name
+                }-icon`}
+                className="mb-6 flex flex-col	items-center"
```
Comment (laurawong) [inline src/components/NavBar/NavBar.jsx]: I would add a class here. `icon {isActive ? "active" : "inactive"}`

**Review by laurawong — APPROVED** [review]

**Pull request merged**